auth/service.cc

```diff
@@ -627,4 +633,64 @@ future<std::vector<permission_details>> list_filtered_permissions(
     });
 }
 
+future<> migrate_to_auth_v2(cql3::query_processor& qp, ::service::raft_group0_client& g0, start_operation_func_t start_operation_func, abort_source& as) {
+    // FIXME: if this function fails it may leave partial data in the new tables
+    // that should be cleared
+    auto gen = [&qp] (api::timestamp_type& ts) -> mutations_generator {
+        for (const auto& cf_name : std::vector<sstring>{
+            "roles", "role_members", "role_attributes", "role_permissions"}) {
+            schema_ptr schema;
+            try {
+                schema = qp.db().find_schema(meta::legacy::AUTH_KS, cf_name);
+            } catch (data_dictionary::no_such_column_family&) {
+                continue; // some tables might not have been created if they were not used
+            }
+            auto rows = co_await qp.execute_internal(
```

Comment:
What timeout is this using? Hopefully not infinite.
Maybe we should pass `internal_distributed_query_state`?

Reply:
default is 1h, on the other hand internal_distributed_query_state is 5s. I'm not sure whether 5s is fine in all cases? in theory it's a select on the whole table

Reply:
CL=ALL writes could timeout if connection is broken to one of the replicas, but the network problem could be transient, so it's better to restart quickly in such situation. (IIUC the procedure retries on failure)

OTOH we don't want the timeout to be too small because as you noticed, it's a whole table select

I suggest setting the timeout to a few minutes, let's say 5 minutes? It should be a good balance.

Reply:
ok